### PR TITLE
remove listener helper code, channels and multiple goroutines

### PR DESCRIPTION
**Pull request by itswisdomagain**

### CODE REVIEW

File: ui/page/components/wallet_account_selector.go [hunk 1]
```diff
@@ -334,56 +331,46 @@ func (ws *WalletAndAccountSelector) setWalletLogo(gtx C) D {
...
-		return
+// ListenForTxNotifications listens for transaction and block updates and
+// updates the selector modal, if the modal is open at the time of the update.
+// The tx update listener MUST be unregistered using ws.StopTxNtfnListener()
+// when the page using this WalletAndAccountSelector widget is exited.
+func (ws *WalletAndAccountSelector) ListenForTxNotifications(window app.WindowNavigator) {
+	txAndBlockNotificationListener := &sharedW.TxAndBlockNotificationListener{
+		OnTransaction: func(transaction *sharedW.Transaction) {
+			// refresh wallets/Accounts list when new transaction is received
+			if ws.selectorModal != nil {
```
Comment (ukane-philemon) [inline ui/page/components/wallet_account_selector.go:342]: I think we can do `if ws.selectorModal == nil { return }` and reduce the nested ifs.

File: ui/page/components/wallet_account_selector.go [hunk 2]
```diff
@@ -334,56 +331,46 @@ func (ws *WalletAndAccountSelector) setWalletLogo(gtx C) D {
...
+					ws.selectorModal.setupWallet()
+				}
+				window.Reload()
+			}
+		},
+		OnBlockAttached: func(walletID int, blockHeight int32) {
+			// refresh wallet and account balance on every new block
+			// only if sync is completed.
+			if ws.selectedWallet.IsSynced() {
+				if ws.selectorModal != nil {
```
Comment (ukane-philemon) [inline ui/page/components/wallet_account_selector.go:355]: Same as my comment above `if !ws.selectedWallet.IsSynced() ||  ws.selectorModal == nil { return }`.

File: ui/page/exchange/create_order_page.go [hunk 3]
```diff
@@ -1297,40 +1283,41 @@ func (pg *CreateOrderPage) updateExchangeConfig() {
...
+	// TODO: add rate listener.
+}
 
-				return
-			}
-		}
-	}()
+func (pg *CreateOrderPage) stopNtfnListeners() {
+	pg.WL.AssetsManager.InstantSwap.RemoveNotificationListener(CreateOrderPageID)
+	// TODO: remove rate listener.
```
Comment (ukane-philemon) [inline ui/page/exchange/create_order_page.go:1322]: Can remove. The rate listener is no longer on this page.

File: ui/page/privacy/account_mixer_page.go [hunk 4]
```diff
@@ -29,8 +27,9 @@ type AccountMixerPage struct {
 	// helper methods for accessing the PageNavigator that displayed this page
 	// and the root WindowNavigator.
 	*app.GenericPageModal
-	*listeners.AccountMixerNotificationListener
-	*listeners.TxAndBlockNotificationListener
+
+	accountMixerNotificationListener *dcr.AccountMixerNotificationListener
+	txAndBlockNotificationListener   *sharedW.TxAndBlockNotificationListener
```
Comment (ukane-philemon) [inline ui/page/privacy/account_mixer_page.go:33]: With the pattern adopted in this PR, I think these should also go. We don't need to keep track of them.

File: libwallet/internal/politeia/politeia_sync.go [hunk 5]
```diff
@@ -575,61 +576,58 @@ func (p *Politeia) CastVotes(ctx context.Context, wallet *wallet.Wallet, eligibl
...
+	if _, ok := p.syncCallbacks[uniqueIdentifier]; ok {
 		return errors.New(ErrListenerAlreadyExist)
 	}
 
-	p.notificationListeners[uniqueIdentifier] = notificationListener
+	p.syncCallbacks[uniqueIdentifier] = syncCallback
 	return nil
 }
 
 func (p *Politeia) RemoveNotificationListener(uniqueIdentifier string) {
```
Comment (ukane-philemon) [inline libwallet/internal/politeia/politeia_sync.go:591]: `RemoveSyncCallback`?

**Review by ukane-philemon — CHANGES_REQUESTED** [review]
Good work! 

I still see this error...  To reproduce, navigate back and forth between the wallet selection page and the selected wallet page.

```
2023-11-06 23:32:28.887 [ERR] UI: No rate information available
2023-11-06 23:32:30.870 [ERR] UI: Error adding sync progress listener: listener_already_exist
2023-11-06 23:32:43.719 [ERR] UI: No rate information available

```

File: ui/page/root/main_page.go [hunk 6]
```diff
@@ -122,21 +111,21 @@ func (mp *MainPage) OnNavigatedTo() {
 	}
 
 	if mp.CurrentPage() == nil {
-		mp.Display(info.NewInfoPage(mp.Load)) // TODO: Should pagestack have a start page?
+		mp.Display(info.NewInfoPage(mp.Load)) // TODO: Should pagestack have a start page? YES!
```
Comment (itswisdomagain) [inline ui/page/root/main_page.go:114]: @ukane-philemon would you like to work on this?

Comment (ukane-philemon) [reply ui/page/root/main_page.go:114]: Yes. I have just 2 PRs under review.

File: ui/page/root/wallet_list.go [hunk 7]
```diff
@@ -351,7 +351,7 @@ func (pg *WalletSelectorPage) listenForSyncProgressNotifications() {
 	for _, w := range allWallets {
 		err := w.AddSyncProgressListener(syncListener, WalletSelectorPageID)
 		if err != nil {
-			log.Errorf("Error adding sync progress listener: %v", err)
+			log.Errorf("3. Error adding sync progress listener: %v", err)
```
Comment (ukane-philemon) [inline ui/page/root/wallet_list.go:354]: We can use the method receiver's name as a prefix so we don't have to number everywhere, or was this just for debugging?

**Review by ukane-philemon — APPROVED** [review]
LGTM. The dup note listener error is fixed.

File: ui/page/root/wallet_list.go (hunk 7 above)
Comment (itswisdomagain) [reply ui/page/root/wallet_list.go:354]: Debugging. Will remove.

**Review by dreacot — APPROVED** [review]

**Pull request merged**